runtime/onert/backend/cpu/ops/FullyConnectedLayer.cc

```diff
@@ -100,6 +101,44 @@ void FullyConnectedLayer::fullyConnectedHybrid()
       getTensorShape(_weights), reinterpret_cast<const int8_t *>(_weights->buffer()),
       getTensorShape(_bias), reinterpret_cast<const float *>(_bias ? _bias->buffer() : nullptr),
       getTensorShape(_output), reinterpret_cast<float *>(_output->buffer()), temp_arena);
+#ifdef USE_RUY_GEMV
+  // very dependent on ruy
+  if (_freed_weights)
+    return;
+
+  const int rows = getTensorShape(_weights).Dims(0);
+  if (rows % 4 != 0)
+  {
+    _freed_weights = true;
+    return;
+  }
+
+  const int total_input_size = getTensorShape(_input).FlatSize();
+  const int input_size = getTensorShape(_weights).Dims(1);
+  const int batch_size = total_input_size / input_size;
+  if (batch_size > 4)
+  {
+    _freed_weights = true;
+    return;
+  }
+
+  // TODO Remove const_cast
+  auto weight_tensor = dynamic_cast<const Tensor *>(_weights);
+  if (weight_tensor)
+  {
+    // TODO Remove this workaround
+    // buffer will have deallocated memory address because ruy kernel uses
+    // it as cache key
+    auto tensor = const_cast<Tensor *>(weight_tensor);
+    auto buffer = tensor->buffer();
```

Comment:
To remove the workaround, can we make `buffer` to be a member of `FullyConnectedLayer`?

Reply:
Oh, that's a good idea. I'll try. Thanks.